GTScript-Parallel-model.md

```diff
@@ -134,17 +186,119 @@ The computation domain of every statement is extended to ensure that any require
 On an applied example, this means:
 
 ```python
-with computation(PARALLEL), interval(...):
+with computation(...), interval(...):
     u = 1
     b = u[-2, 0, 0] + u[1, 0, 0] + u[0, -1, 0] + u[0, -2, 0]
 ```
 
 translates into the following pseudo code:
 
+```python
+for k in range(k, end):
+    u[i-2:J+1, j-2:J, k] = 1
+    b[i:I, j:J, k] = u[i-2:I-2, j:J, k] + u[i+1:I+1, j:J, k] + u[i:I,j-1:J-1,k] + u[i:I,j-2:J-2,k]
+```
+
+## Conditionals
+
+GTScript supports 2 kinds of conditionals:
+
+- conditionals on scalar expressions
+- conditionals on field expressions
+
+### Conditionals on scalar expressions
+
+- Each statement inside the if and else branches is executed according to the same rules as statements outside of branches.
+- There is no restriction on the body of the statement.
+
+**Example for scalar conditions**
+
+```python
+with computation() with interval(...):
+    if my_config_var:
+        a = 1
+        b = 2
+    else:
+        a = 2
+        b = 1
+```
+
+translates to:
+
+```python
+for k in range(k, K):
+    parfor ij:
+        if my_config_var:
+            a[i, j, k] = 1
+    parfor ij:
+        if my_config_var:
+            b[i, j, k] = 2
+    parfor ij:
+        if not my_config_var:
+            a[i, j, k] = 2
+    parfor ij:
+        if not my_config_var:
+            b[i, j, k] = 1
+```
+
+### Conditionals on field expressions
+
+- The condition is evaluated for all gridpoints and stored in a mask.
+- Each statement inside the if and else branches is executed according to the same rules as statements outside of branches.
+- Inside the if and else blocks the same field cannot be written to **and** read with an offset in the parallel axises (order does not matter).
```

Comment:
```suggestion
- Inside the if and else blocks the same field cannot be written to **and** read with an offset in the `I`,`J` parallel axes (order does not matter).
```
Actually, we never say that the I,J horizonal axes are parallel and K is not. Should we extend the domain description at the beginning of the document?